src/std/misc/text.ss

```diff
@@ -13,3 +22,169 @@
      (let (path (gx#core-resolve-path #'path (stx-source stx)))
        (with-syntax ((txt (call-with-input-file path read-all-as-string)))
          #'(quote txt))))))
+
+;; quasistring utility functions
+(begin-syntax
+  (def qs-err-missing-close "Malformed quasistring: closing } required after s-exp")
+
+  (def (stx-format stx)
+    (if (stx-string? stx) ;; don't format strings
+      stx
+      (with-syntax ((stx stx))
+        #'(format "~a" stx))))
+
+  (def (map-format stx)
+    (stx-map stx-format stx))
+
+  ;; custom eval to capture the proper variables
+  (def (eval** stx ctx)
+    (parameterize ((current-expander-phi (1+ (current-expander-phi))))
+      (let (stx (datum->syntax ctx stx))
+        (eval-syntax stx #t))))
+
+  (def ht (datum->syntax #f 'ht)) ;; hash table placeholder
+  (def al (datum->syntax #f 'al)) ;; alist placeholder
+  (def pl (datum->syntax #f 'pl)) ;; plist placeholder
+
+  (def (parse-quasistring stx ctx rt-symbol-proc output-proc (expansion-time-eval #f))
+    ;; rt-symbol-proc wraps sexp if sexp is a run time symbol
+    ;; output-proc is what we return from the macro
+    ;; if expansion-time-eval is #t, allow for ##{} template variables
+    (def (lp (port (open-input-string (stx-e stx)))
+             (c    (read-char port))
+             (ops  [])  ;; arguments of string-append
+             (str  "")) ;; container to concatenate strings
+      (cond
+       ((eq? c #\#)
+        (let (c+1 (read-char port))
+          (cond
+           ((eq? c+1 #\{) ;; run time template variables
+            (let ((sexp (read port))
+                  (c+2  (read-char port)))
+              (if (eq? c+2 #\})
+                (if (symbol? sexp)
+                  (lp port (read-char port) (append ops [str (rt-symbol-proc sexp)]) "")
+                  (lp port (read-char port) (append ops [str sexp]) ""))
+                (error qs-err-missing-close c+2))))
+           ((eq? c+1 #\#) ;; expansion time template variables
+            (if expansion-time-eval
```

Comment:
shouldn't we throw an error if `expanstion-time-eval` is `#f` here?

Reply:
I don't think so. I want this behaviour:

```scheme
> (let (name "world") (quasistring "Hello, #{name}! ##{name}"))
"Hello, world! ##{name}"
```
I don't want to have two parsers. It can definitely be made better but I don't think throwing an error is what we want.